flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/HybridShuffleTestUtils.java

```diff
@@ -73,37 +80,96 @@ public static HsOutputMetrics createTestingOutputMetrics() {
-                numBuffersPerRegion,
-                new boolean[numBuffersPerRegion]);
+        return new TestingFileDataIndexRegion.Builder()
+                .setGetSizeSupplier(() -> TestingFileDataIndexRegion.REGION_SIZE)
+                .setContainBufferFunction(
+                        bufferIndex ->
+                                containBufferFunction(
+                                        bufferIndex, firstBufferIndex, numBuffersPerRegion))
+                .setGetFirstBufferIndexSupplier(() -> firstBufferIndex)
+                .setGetRegionFileOffsetSupplier(() -> firstBufferOffset)
+                .setGetNumBuffersSupplier(() -> numBuffersPerRegion)
+                .build();
     }
 
-    public static List<InternalRegion> createAllUnreleasedRegions(
+    public static List<TestingFileDataIndexRegion> createTestRegions(
             int firstBufferIndex, long firstBufferOffset, int numBuffersPerRegion, int numRegions) {
-        List<InternalRegion> regions = new ArrayList<>();
+        List<TestingFileDataIndexRegion> regions = new ArrayList<>();
         int bufferIndex = firstBufferIndex;
         long bufferOffset = firstBufferOffset;
+        int numRegionSize = TestingFileDataIndexRegion.REGION_SIZE;
         for (int i = 0; i < numRegions; i++) {
+            final int currentBufferIndex = bufferIndex;
+            final long currentBufferOffset = bufferOffset;
             regions.add(
-                    new InternalRegion(
-                            bufferIndex,
-                            bufferOffset,
-                            numBuffersPerRegion,
-                            new boolean[numBuffersPerRegion]));
+                    new TestingFileDataIndexRegion.Builder()
+                            .setGetSizeSupplier(() -> numRegionSize)
+                            .setGetFirstBufferIndexSupplier(() -> currentBufferIndex)
+                            .setGetRegionFileOffsetSupplier(() -> currentBufferOffset)
+                            .setGetNumBuffersSupplier(() -> numBuffersPerRegion)
+                            .setContainBufferFunction(
+                                    index ->
+                                            containBufferFunction(
+                                                    index, firstBufferIndex, numBuffersPerRegion))
+                            .build());
             bufferIndex += numBuffersPerRegion;
             bufferOffset += bufferOffset;
         }
         return regions;
     }
 
-    public static void assertRegionEquals(InternalRegion expected, InternalRegion region) {
+    public static FileDataIndexRegionHelper.Region createSingleFixedSizeRegion(
+            int firstBufferIndex, long firstBufferOffset, int numBuffersPerRegion) {
+        return new ProducerMergedPartitionFileIndex.FixedSizeRegion(
+                firstBufferIndex, firstBufferOffset, numBuffersPerRegion);
+    }
+
+    public static TestingFileDataIndexRegion readRegionFromFile(
```

Comment:
Maybe we can put these two method to `TestingFileDataIndexRegion.class`.

Reply:
They are moved to `TestingFileDataIndexRegion `